test/integration-tests/V3ToV4Migration.test.ts

```diff
@@ -404,6 +404,43 @@ describe('V3 to V4 Migration Tests:', () => {
 
         await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
       })
+
+      it('eve permits bob for the token and router for all tokens - he can call decrease even though he is not the owner', async () => {
+        // transfer the token to eve
+        await v3NFTPositionManager.transferFrom(bob.address, eve.address, tokenIdv3)
+
+        // eve approves the router to spend all of her tokens
+        await v3NFTPositionManager.connect(eve).setApprovalForAll(router.address, true)
+
+        // eve creates a signature for bob to spend the token
+        let { v, r, s } = await getPermitNFTSignature(eve, v3NFTPositionManager, bob.address, tokenIdv3, MAX_UINT)
+        const erc721PermitParams = {
+          spender: bob.address,
+          tokenId: tokenIdv3,
+          deadline: MAX_UINT,
+          v: v,
+          r: r,
+          s: s,
+        }
+
+        const encodedErc721PermitCall = encodeERC721Permit(erc721PermitParams)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [encodedErc721PermitCall])
+
+        await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+        planner = new RoutePlanner()
+
+        let position = await v3NFTPositionManager.positions(tokenIdv3)
+        let liquidity = position.liquidity
+
+        const params = { tokenId: tokenIdv3, liquidity: liquidity, amount0Min: 0, amount1Min: 0, deadline: MAX_UINT }
+
+        const encodedDecreaseCall = encodeDecreaseLiquidity(params)
+
+        planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedDecreaseCall])
+
+        await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
```

Comment:
does this need to be 2 calls to `executeRouter`? can it just be 1?

Reply:
oh yeah changed the one above it too